src/classic/clvm/casts.rs

```diff
@@ -107,89 +106,39 @@ pub fn bigint_from_bytes(b: &Bytes, option: Option<TConvertOption>) -> Number {
     unsigned
 }
 
-pub fn bigint_to_bytes(v_: &Number, option: Option<TConvertOption>) -> Result<Bytes, String> {
-    let v = v_.clone();
-
-    if v == bi_zero() {
+// Convert to bytes with no leading zeros added.  This allows numbers to appear
+// negative in the result stream but are treated as positive (for node paths).
+pub fn bigint_to_bytes_unsigned(v: &Number) -> Result<Bytes, String> {
```

Comment:
I don't see unit tests for this function either. having tests demonstrating the difference between this and `bigint_to_bytes_clvm` would be nice.

on an unrelated note, I'm a bit surprised to see `Bytes` be an optional type, and that an empty array is represented as `None`. Is this an artifact from a direct translation from Python or JS?